hpedockerplugin/volume_manager.py

```diff
@@ -434,19 +515,332 @@ def get_path(self, volname):
         response = json.dumps({u"Err": '', u"Mountpoint": path_name})
         return response
 
+    @staticmethod
+    def _is_vol_not_mounted(vol):
+        return 'node_mount_info' not in vol
+
+    @staticmethod
+    def _is_first_mount(node_mount_info):
+        return (len(node_mount_info) == 0)
+
+    def _is_vol_mounted_on_this_node(self, node_mount_info):
+        return self._node_id in node_mount_info
+
+    def _update_mount_id_list(self, vol, mount_id):
+        LOG.debug("NODE_MOUNT_INFO - NOT first mount ID - APPENDING "
+                  "%s" % mount_id)
+        node_mount_info = vol['node_mount_info']
+        node_mount_info[self._node_id].append(mount_id)
+        LOG.debug("NODE_MOUNT_INFO - NOT first mount ID - UPDATING "
+                  "ETCD %s" % mount_id)
+        self._etcd.update_vol(vol['id'],
+                              'node_mount_info',
+                              node_mount_info)
+
+    def _get_success_response(self, vol):
+        path_info = self._etcd.get_vol_path_info(vol['display_name'])
```

Comment:
@imran-ansari  @wdurairaj Can we use path_info = json.loads(vol['path_info']) this will help for snapshot mount operation as well , else it restricts its operation on parent vol attributes and return parents path information only.

Reply:
Okay